src/Microsoft.NetCore.Analyzers/Core/Runtime/DisposableFieldsShouldBeDisposed.cs

```diff
@@ -77,45 +83,54 @@ void addOrUpdateFieldDisposedValue(IFieldSymbol field, bool disposed)
                         {
-                            Debug.Assert(disposeAnalysisResult != null);
-                            Debug.Assert(pointsToAnalysisResult != null);
+                            var fieldReference = (IFieldReferenceOperation)operationContext.Operation;
+                            var field = fieldReference.Field;
 
-                            operationBlockStartContext.RegisterOperationAction(operationContext =>
+                            // Only track instance fields on the current instance.
+                            if (field.IsStatic || fieldReference.Instance?.Kind != OperationKind.InstanceReference)
                             {
-                                var fieldReference = (IFieldReferenceOperation)operationContext.Operation;
-                                var field = fieldReference.Field;
+                                return;
+                            }
 
-                                // Only track instance fields on the current instance.
-                                if (field.IsStatic || fieldReference.Instance?.Kind != OperationKind.InstanceReference)
-                                {
-                                    return;
-                                }
+                            // Check if this is a Disposable field that is not currently being tracked.
+                            if (fieldDisposeValueMap.ContainsKey(field) ||
+                                !disposeAnalysisHelper.GetDisposableFields(field.ContainingType).Contains(field))
+                            {
+                                return;
+                            }
 
-                                // Check if this is a Disposable field that is not currently being tracked.
-                                if (fieldDisposeValueMap.ContainsKey(field) ||
-                                    !disposeAnalysisHelper.GetDisposableFields(field.ContainingType).Contains(field))
+                            // We have a field reference for a disposable field.
+                            // Check if it is being assigned a locally created disposable object.
+                            if (fieldReference.Parent is ISimpleAssignmentOperation simpleAssignmentOperation &&
+                                simpleAssignmentOperation.Target == fieldReference)
+                            {
+                                if (lazyPointsToAnalysisResult == null)
                                 {
-                                    return;
+                                    if (disposeAnalysisHelper.TryGetOrComputeResult(operationBlockStartContext.OperationBlocks,
+                                        containingMethod, operationBlockStartContext.Options, Rule, trackInstanceFields: false,
+                                        trackExceptionPaths: false, operationBlockStartContext.CancellationToken,
+                                        out var disposeAnalysisResult, out var pointsToAnalysisResult))
```

Comment:
Could you get the same PointsToAnalysisResult that DisposeAnalysis, from PointsToAnalysis's GetOrComputeResult()?  Then could avoid an extra DFA pass of dispose analysis here.

Reply:
That is true - I was just being lazy. Let me replace this with just points to computation.

Reply:
Done, thanks!